velox/common/encode/tests/EncoderUtilsTests.cpp

```diff
@@ -0,0 +1,168 @@
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 62,  255,
+    62,  255, 63,  52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  255, 255,
+    255, 255, 255, 255, 255, 0,   1,   2,   3,   4,   5,   6,   7,   8,   9,
+    10,  11,  12,  13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,
+    25,  255, 255, 255, 255, 63,  255, 26,  27,  28,  29,  30,  31,  32,  33,
+    34,  35,  36,  37,  38,  39,  40,  41,  42,  43,  44,  45,  46,  47,  48,
+    49,  50,  51,  255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+    255};
+
+int base64 = 64;
+
+TEST_F(EncoderUtilsTest, baseReverseLookup) {
+  EXPECT_NO_THROW(baseReverseLookup(base64, 'A', testReverseIndex));
+  EXPECT_THROW(
+      baseReverseLookup(base64, '=', testReverseIndex), VeloxUserError);
+}
+
+TEST_F(EncoderUtilsTest, checkForwardIndex) {
+  EXPECT_TRUE(checkForwardIndex(63, testCharset, testReverseIndex));
+}
+
+TEST_F(EncoderUtilsTest, checkReverseIndex) {
+  EXPECT_TRUE(checkReverseIndex(255, base64, testCharset, testReverseIndex));
+}
+
+TEST_F(EncoderUtilsTest, HandlesLookupAndExceptions) {
+  EXPECT_NO_THROW(baseReverseLookup(base64, 'A', testReverseIndex));
+  EXPECT_THROW(
+      baseReverseLookup(base64, '=', testReverseIndex), VeloxUserError);
+}
+
+TEST_F(EncoderUtilsTest, ValidatesCharsetWithReverseIndex) {
+  EXPECT_TRUE(checkForwardIndex(63, testCharset, testReverseIndex));
+}
+
+TEST_F(EncoderUtilsTest, ValidatesReverseIndexWithCharset) {
+  EXPECT_TRUE(checkReverseIndex(
+      sizeof(testCharset) - 1, base64, testCharset, testReverseIndex));
+  EXPECT_FALSE(checkReverseIndex(
+      sizeof(testWrongCharset) - 1,
+      base64,
+      testWrongCharset,
+      testWrongReverseIndex));
+}
+
+TEST_F(EncoderUtilsTest, CharacterDoesNotExist) {
+  Charset charset = {'A', 'B', 'C', 'D', 'E', 'F'};
+  EXPECT_TRUE(findCharacterInCharSet(charset, charset.size(), 1, 'C'));
+  EXPECT_FALSE(findCharacterInCharSet(charset, charset.size(), 1, 'A'));
+  EXPECT_TRUE(findCharacterInCharSet(charset, charset.size(), 0, 'A'));
+}
+
+TEST_F(EncoderUtilsTest, EmptyCharset) {
+  Charset emptyCharset;
+  EXPECT_FALSE(
+      findCharacterInCharSet(emptyCharset, base64, emptyCharset.size(), 'A'));
+}
+
+TEST_F(EncoderUtilsTest, CheckForwardIndex_PartialValid) {
```

Comment:
These are technically class and method names so would fall under that syntax.
How about `partialValidWithForwardIndex` and similar for the others?